### ISSUE TITLE
Clean up keyboard focus in search filters

### ISSUE THREAD

**Issue by szakarias** [dart-lang/pub-dev#7282]
There should not be a keyboard focus both on the checkbox and the text next to the check box. A single focus should be enough.

As it is now when tabbing through the site, keyboard focus is first on the checkbox, but it is not visible. Here it can be checked with a 'space' button. Next tab puts focus on the text, and here the checkbox can be checked the 'enter' button. 

We could consider only being able to check the box with space.

![image](https://github.com/dart-lang/pub-dev/assets/3865266/4bb5e0f5-553d-40bd-b787-129f916a02c4)

**Comment by sigurdm** [comment]
As a bonus feature we should consider keeping the focus on the page-refresh after the settings have been updated.
